client/config.c

```diff
@@ -143,13 +145,24 @@ TDNFReadConfig(
                   pTdnf);
     BAIL_ON_TDNF_ERROR(dwError);
 
+    dwError = TDNFAllocateString(pszConfFile, &pszConfFileCopy);
```

Comment:
do we need pszConfFileCopy? we arent adding anything to pszConfFile, right?

Reply:
We use `dirname` on it in line 151, which modifies the argument. We don't want to mess up `pszConfFile` which was received in this function as an argument. I can add a comment.